src/clients/UBAClient/UBAClientBase.ts

```diff
@@ -15,6 +14,10 @@ import { computeLpFeeStateful } from "./UBAClientUtilities";
 import { findLast } from "../../utils/ArrayUtils";
 import { analog } from "../../UBAFeeCalculator";
 import { BaseAbstractClient } from "../BaseAbstractClient";
+import { HubPoolClient } from "../HubPoolClient";
+import { SpokePoolClients } from "../../utils";
+import { ERC20__factory } from "../../typechain";
+import { TOKEN_SYMBOLS_MAP } from "@across-protocol/contracts-v2";
```

Comment:
The goal of this class was to not use any RPC endpoint calls and rely entirely on a state object for the Vercel API. Do you recommend that we keep the original method or should we deviate?

Reply:
I see what you're saying. So original issue I'm trying to fix with `computeLpFee` is that you're using the incorrect `ethSpokeBalance`, `hubBalance`, and `hubEquity`. Those values need to be linked to the passed in `hubPoolBlockNumber`, not the values associated with the `bundleEndBlock` of the latest bundle state returned by `retrieveLastBundleState`.

So one way to make this stateless is to require the caller to pass in `ethSpokeBalance`, `hubBalance`, and `hubEquity`.